DrawerKit/DrawerKit/Public API/Notifications.swift

```diff
@@ -12,13 +12,38 @@ public class NotificationToken {
     deinit { center.removeObserver(token) }
 }
 
+@available(*, deprecated, message: "Do not use this.")
 public protocol NotificationEnum {
     var name: Notification.Name { get }
 }
 
 extension NotificationCenter {
     private static let notificationKey = "notificationKey"
 
+    public func addObserver<T: DrawerNotificationInfo>(_ type: T.Type,
```

Comment:
### Note

Here I adopted explicit type argument for better type-inference, just like `Codable`.